MQuince/MQuince.WebAPI.Integration.Testing/AppointmentTests.cs

```diff
@@ -79,6 +80,27 @@ public async Task Create_appointments()
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
         }
 
+        [Theory]
+        [MemberData(nameof(AppointmentData))]
+        public async void Cancel_appointment_status_code_test(Guid appointmentId, HttpStatusCode expectedResponseStatusCode)
+        {
+            HttpClient client = _factory.CreateClient();
+
+            var response = await client.PutAsync("/api/Appointment/cancelAppointment/" + appointmentId, new StringContent("1", Encoding.UTF8, "application/json"));
+
+            response.StatusCode.ShouldBeEquivalentTo(expectedResponseStatusCode);
+        }
+
+        public static IEnumerable<object[]> AppointmentData()
+        {
+            var retVal = new List<object[]>();
+            Guid appointmentId1 = Guid.Parse("08d8a3a6-5fcd-424e-85ce-d5276d99b442");
+            Guid appointmentId2 = Guid.Parse("08d8a2ab-e138-4e36-8542-68bec07b074c");
+            retVal.Add(new object[] { appointmentId1, HttpStatusCode.OK });
+            retVal.Add(new object[] { appointmentId2, HttpStatusCode.BadRequest });
+            return retVal;
+        }
+
```

Comment:
Your controller have 4 different scenarios (returns 4 different status code depending on situation). You should have at least 4 tests for this.